src/app/pages/surveillance/surveillance/components/view.component.js

```diff
@@ -5,10 +5,11 @@ export const SurveillanceManagementViewComponent = {
         takeAction: '&',
     },
     controller: class SurveillanceManagementViewComponent {
-        constructor ($log, utilService) {
+        constructor ($log, authService, utilService) {
             'ngInject'
             this.$log = $log;
             this.certificationStatus = utilService.certificationStatus;
+            this.canEdit = authService.hasAnyRole(['ROLE_ADMIN', 'ROLE_ONC']);
```

Comment:
This sets the boolean value of "canEdit" to a fixed value at the time that the surveillance is loaded. If the user is not logged in at that time, but logs in later, it won't change that value to allow them to edit. Usually, in this case, I'd make this line 
```suggestion
            this.hasAnyRole = authService.hasAnyRole;
```
and add a function below like

```
canEdit () {
return this.hasAnyRole(['ROLE_ADMIN', 'ROLE_ONC']) ||
(this.hasAnyRole(['ROLE_ACB']) && this.listing.certificationEdition.name !== '2014');
}
```

Then, in the HTML, the code would be
```
allow-editing="$ctrl.canEdit()"
```